### PR TITLE
Add support for Azure, OpenAI, Palm, Anthropic, Cohere, Replicate Models - using litellm

### PR DESCRIPTION
**Pull request by ishaan-jaff**
## Problem

Adding support for Azure, Claude-2, Llama2, Palm LLM models. Claude-2 has larger context windows and IMO is better at generating SQL queries. instead of needing to leave posthog to use it, i'd rather us it on the product. 

## Changes
This PR adds support for models from all the above mentioned providers using litellm https://github.com/BerriAI/litellm - a simple & light package to call OpenAI, Azure, Cohere, Anthropic API Endpoints

Here's a sample of how it's used:

```
from litellm import completion

## set ENV variables
# ENV variables can be set in .env file, too. Example in .env.example
os.environ["OPENAI_API_KEY"] = "openai key"
os.environ["COHERE_API_KEY"] = "cohere key"

messages = [{ "content": "Hello, how are you?","role": "user"}]

# openai call
response = completion(model="gpt-3.5-turbo", messages=messages)

# cohere call
response = completion("command-nightly", messages)

# anthropic call
response = completion(model="claude-instant-1", messages=messages)
```

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?
Ran locally and ensured all liteLLM supported model calls worked

### CODE REVIEW

**Comment by ishaan-jaff** [comment]
@liyiy @Twixes please can you take a look at this PR when you get the chance? 

If the initial commit looks good, happy to add docs/tests on the same😊

**Review by Twixes — COMMENTED** [review]
Hi @ishaan-jaff! This is definitely an interesting contribution, `litellm` seems super handy. At the moment we're not actively working on deploying the AI-assisted SQL experience publicly, so I'm going to leave this open for some time until we come back to it. Your library may be helpful in trialing Claude Instant instead of GPT-3.5 (although probably not Claude 2 due to much higher cost). We don't plan to support multiple AI providers at once though, so to be honest it's unlikely we'll use `litellm` in production.

**Comment by ishaan-jaff** [comment]
@Twixes thanks for the feedback! sounds good

**Comment by ghost** [comment]
@Twixes wouldn't you do model fallbacks in prod - so you'd still probably be switching b/w llms in prod - no?

**Comment by Twixes** [comment]
@krrishdholakia Likely not, though I'm curious to read more up on this, if you have some literature :)

**Comment by ishaan-jaff** [comment]
@Twixes here are our docs on using fallbacks with LiteLLM: https://docs.litellm.ai/docs/completion/reliable_completions 

**TLDR:** 
Hit a context window error gpt-3.5 -> retry request on gpt-3.5-16k
Hit a rate limit error on gpt-4 -> fallback the request to gpt-3.5, claude, cohere etc 
Hit a rate limit on a an Azure deployment -> fallback to other azure deployments

**Comment by mariusandra** [comment]
@ishaan-jaff @krrishdholakia your project looks amazing! Yet at this moment we really are not investing in LLM-based solutions. Thus I'd strongly prefer to not add an unused dependency into `requirements.txt`, and increase our maintenance burden. I hope you understand.

That said, we will likely revisit this when we work on features to e.g. automatically summarise surveys, etc. @neilkakkar @liyiy - keep this on your radar.

**Comment by ghost** [comment]
Thanks @mariusandra,.

@neilkakkar @liyiy I would **love** to make [LiteLLM](https://github.com/BerriAI/litellm) a great product for y'all. 

Please let me know if you feel like anything is missing from the project.

Also reached out via linkedin - if it's more convenient to chat there.
